egs/wsj/s5/steps/nnet3/components.py

```diff
@@ -347,4 +347,164 @@ def AddClstmLayer(config_lines,
             }
 
 
+def AddBlstmLayer(config_lines,
```

Comment:
Added. One thing:

in your example:
y(t) = Wyr * r(t) + Wyp * p(t) + by

But according to the existing AddLstmLayer, it seems that it is
y(t) = [r_forward(t); p_forward(t)]
i.e. there is no further affine transforms  Wyr and Wyp on r(t) and p(t) respectively.

SInce my Blstm is built on the the exsisting Lstm, I also wrote y(t) in the Append form.